backend/src/main/java/com/google/sps/data/Match.java

```diff
@@ -1,34 +1,27 @@
 package com.google.sps.data;
 
-/**
- * A match of two participants. Note: Match includes entire Participants (with unique ID per
- * submitted form) instead of just usernames because a username may be involved in multiple matches
- * over time. This issue can be resolved if we delete matches from the datastore after the users are
- * notified of the match.
- */
+import com.google.appengine.api.datastore.Key;
+
+/** A match between two participants. */
 public final class Match {
 
-  /** Datastore entity ID */
+  /** Match datastore key ID */
   private final long id;
-  /** First participant */
-  private final Participant firstParticipant;
-  /** Second participant */
-  private final Participant secondParticipant;
+  /** First participant datastore key */
+  private final Key firstParticipantKey;
+  /** Second participant datastore key */
+  private final Key secondParticipantKey;
```

Comment:
This change breaks the abstraction of the new *store classes. Ideally those classes entirely encapsulate interactions with appengine.api.datastore and so passing datastore keys around in these model classes ties them to a specific datastore implementation. With perfect encapsulation, one could refactor the *store classes within this PR to use something other than appengine.api.datastore for storage without affecting any other code.

Reply:
The problem I ran into is that a Participant is an invalid property type (only types like int, long, Key, etc. can be stored). An alternative is to have a Match consist of actual Participants, but only store the Participant Keys in the datastore. Then MatchDatastore would depend on ParticipantDatastore to switch between Match objects and Match entities. Do you think that would be better?

Reply:
I like the idea of pulling Participant out of this class. Match and Participant are two separate models that are stored separately. While Match has to reference the matched participants, its model doesn't necessarily need to contain the full Participant object. 

This is where you would introduce String keys or long IDs for your Participants that aren't tied to the storage implementation. User emails make sense as keys to me.

Reply:
I changed Key references to be String usernames for each Match, and for each participant, the match reference is now just the long id.